src/devices/machine/wd_fdc.cpp

```diff
@@ -1683,6 +1683,15 @@ bool wd_fdc_device_base::read_one_bit(const attotime &limit)
 	return false;
 }
 
+void wd_fdc_device_base::extract_data_reg()
+{
+	cur_live.data_reg = 0;
+	for (int bit = 0; bit < 8; bit++) {
+		if (cur_live.shift_reg & (1<<(bit*2)))
+			cur_live.data_reg |= (1<<bit);
+	}
```

Comment:
MAME has a few utility functions to help with bit operations. Since this is just grabbing the even bits one possibility is:
```cur_live.data_reg = bitswap<8>(cur_live.shift_reg, 14, 12, 10, 8, 6, 4, 2, 0);```

Reply:
Thank you, I have updated the change to use the template.